hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/StoreFileInfo.java

```diff
@@ -109,6 +111,11 @@
    */
   public StoreFileInfo(final Configuration conf, final FileSystem fs, final Path initialPath)
       throws IOException {
+    this(conf, fs, initialPath, null, null);
+  }
+
+  private StoreFileInfo(final Configuration conf, final FileSystem fs, final Path initialPath,
+      final Long createdTimestamp, final Long size) throws IOException {
```

Comment:
What about just passing a FileStatus in?

Reply:
I initially thought the same Duo like directly passing "FileStatus" in constructor but what I feel that "Path" is also in same constructor and even that Path can be constructed from FileStatus.  So that looks to me not right.  :) Another important is Path can not be removed from constructor because many callers pass only Path object, not FileStatus.

I am fine Duo if you still want to change, 
public StoreFileInfo(final Configuration conf, final FileSystem fs, final Path initialPath, FileStatus filestatus)

Reply:
I think it is fine, as the FileStatus could be null. And maybe you could add a @Nullable annotation to indicate this.

Reply:
Passing the individual params Ts and size looks cleaner IMO. Anyways explained why selected that path.

Reply:
But this requires extra boxing/unboxing? And what if we need more parameters in the future...

Reply:
I did not really notice it is Long not long. Good eyes Duo.. Ya why the wrapped type?

Reply:
So what's your opinion @anoopsjohn ? Keep it as is or change to use FileStatus? Thanks.

Reply:
Thank you very much @anoopsjohn @Apache9.  I used wrapped type because I thought its good choice for a "null" check instead of checking whether StoreFile length is 0 or more. sorry I may be wrong :) I understand Duo concerns likely possible unboxing overhead for each conversion. Let me change.

Reply:
Any way is ok.  As this was not in a hot path, even that Autoboxing thing was also ok.  That 2 params passing looked more clean from the calling stand point.  Anyway not so big concern.  This looks clear too.

Reply:
Thank you @anoopsjohn for the confirmation.  

@Apache9 Duo, I already did new commit for your suggestions. Shall I revert now? 
https://github.com/apache/hbase/pull/674/commits/6f4baaef593cadaaa08904bb8edab8a9876561a2

Reply:
Its ok.. Keep it as the way u have pushed as of now.. Its ok only.

Reply:
Sure @anoopsjohn. Thanks again!!

Reply:
Appreciate Duo @Apache9 if you get some time for review :)